modules/rational-modeline.el

```diff
@@ -0,0 +1,134 @@
+;;;; rational-modeline.el --- rational module for setting up the modeline  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2022
+;; SPDX-License-Identifier: MIT
+
+;; Authors: System Crafters Community, David "System Crafters" Wilson, Erik Lundstedt
+
+;; Commentary
+
+;;
+
+;;; Code:
+
+(defcustom rational-modeline/type "doom"
+  "one of 'GNU' 'doom' 'telephone' ";;'space WIP' "
+  )
```

Comment:
You will need to handle the situation where this changes (e.g. with `:set`), otherwise it will not take effect unless it is configured _before_ the module is required.

`rational-modeline/type` should be `rational-modeline-type` for consistency with style guide.

Reply:
> You will need to handle the situation where this changes (e.g. with :set)
not sure how to do this, happy to get an example or something like that.

I havent really written a lot of elisp code and allmost none outside of my personal configs where this doesnt matter too much

Reply:
> rational-modeline/type should be rational-modeline-type for consistency with style guide.
this should be fixed now



not sure how to add changes to pr... but i fixed this in a recent push

Reply:
> I havent really written a lot of elisp code and allmost none outside of my personal configs where this doesnt matter too much

No worries. One of the Principles of Rational Emacs: "Helps you learn Emacs Lisp"

Reply:
For the choice of options, you should probably use something like the following in your `defcusom` definition:

``` emacs-lisp
    :type '(choice (const :tag "GNU default" gnu)
                   (const :tag "Doom modeline" doom)
                   (const :tag "Telephone modeline" telephone))
```